// @looker/sdk-rtl/lib/esm/authSession.js
function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import { sdkError } from './transport';
export class AuthSession {
  constructor(settings, transport) {
    _defineProperty(this, "settings", void 0);

    _defineProperty(this, "sudoId", '');

    _defineProperty(this, "transport", void 0);

    this.settings = settings;
    this.transport = transport;
  }

  login(_sudoId) {
    return Promise.reject(AuthSession.TBD);
  }

  logout() {
    return Promise.resolve(false);
  }

  isSudo() {
    return this.sudoId !== '' && this.isAuthenticated();
  }

  reset() {
    this.sudoId = '';
  }

  ok(promise) {
    return _asyncToGenerator(function* () {
      var result = yield promise;

      if (result.ok) {
        return result.value;
      } else {
        if (result instanceof Buffer) {
          throw sdkError({
            message: result.toString('utf8')
          });
        } else {
          throw sdkError(result);
        }
      }
    })();
  }

}

_defineProperty(AuthSession, "TBD", 'Method not implemented in AuthSession');
