// @looker/sdk-rtl/lib/esm/constants.js
export var matchCharset = ';.*charset=';
export var matchCharsetUtf8 = "".concat(matchCharset, ".*\\butf-8\\b");
export var matchModeString = "(^application\\/.*(\\bjson\\b|\\bxml\\b|\\bsql\\b|\\bgraphql\\b|\\bjavascript\\b|\\bx-www-form-urlencoded\\b)|^text\\/|.*\\+xml\\b|".concat(matchCharset, ")");
export var matchModeBinary = '^image\\/|^audio\\/|^video\\/|^font\\/|^application\\/|^multipart\\/';
export var isTrue = value => /^(true|t|yes|y|1)$/i.test(value);
export var isFalse = value => /^(false|f|no|n|0)$/i.test(value);
export var boolDefault = function boolDefault(value) {
  var defaultBool = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
  if (isTrue(value)) return true;
  if (isFalse(value)) return false;
  return defaultBool;
};
export var unquote = value => {
  if (!value) return '';

  if (/^['"`]/.test(value)) {
    var quote = value.substring(0, 1);
    if (value.endsWith(quote)) return value.substring(1, value.length - 1);
  }

  return value;
};
