// @looker/sdk-rtl/lib/esm/extensionSession.js
function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

export class ExtensionSession {
  constructor(settings, transport) {
    this.settings = settings;

    _defineProperty(this, "sudoId", '');

    _defineProperty(this, "transport", void 0);

    this.settings = settings;
    this.transport = transport;
  }

  isAuthenticated() {
    return true;
  }

  authenticate(init) {
    return _asyncToGenerator(function* () {
      return Promise.reject(Error("Authenticate ".concat(init ? 'request property overrides' : '', " not supported from ExtensionSession")));
    })();
  }

  getToken() {
    return _asyncToGenerator(function* () {
      return Promise.reject(Error('Access to token is not allowed from ExtensionSession'));
    })();
  }

  isSudo() {
    throw new Error('isSudo is not allowed from ExtensionSession');
  }

  login(sudoId) {
    return _asyncToGenerator(function* () {
      return Promise.reject(Error("Login ".concat(sudoId ? 'for sudo' : '', " not supported from ExtensionSession")));
    })();
  }

  logout() {
    return _asyncToGenerator(function* () {
      return Promise.reject(Error('Logout not supported from ExtensionSession'));
    })();
  }

  reset() {}

}
