// @looker/extension-sdk/lib/esm/connect/types.js
export var ExtensionEvent;

(function (ExtensionEvent) {
  ExtensionEvent["EXTENSION_HOST_NOTIFICATION"] = "EXTENSION_NOTIFICATION";
  ExtensionEvent["EXTENSION_API_REQUEST"] = "EXTENSION_API_REQUEST";
})(ExtensionEvent || (ExtensionEvent = {}));

export var ExtensionRequestType;

(function (ExtensionRequestType) {
  ExtensionRequestType["CONTEXT_DATA"] = "CONTEXT_DATA";
  ExtensionRequestType["VERIFY_HOST"] = "VERIFY_HOST";
  ExtensionRequestType["INVOKE_CORE_SDK"] = "INVOKE_CORE_SDK";
  ExtensionRequestType["RAW_INVOKE_CORE_SDK"] = "RAW_INVOKE_CORE_SDK";
  ExtensionRequestType["UPDATE_TITLE"] = "UPDATE_TITLE";
  ExtensionRequestType["UPDATE_LOCATION"] = "UPDATE_LOCATION";
  ExtensionRequestType["ROUTE_CHANGED"] = "ROUTE_CHANGED";
  ExtensionRequestType["CLOSE_HOST_POPOVERS"] = "CLOSE_HOST_POPOVERS";
  ExtensionRequestType["CLIPBOARD"] = "CLIPBOARD";
  ExtensionRequestType["LOCAL_STORAGE"] = "LOCAL_STORAGE";
  ExtensionRequestType["USER_ATTRIBUTE"] = "USER_ATTRIBUTE";
  ExtensionRequestType["TRACK_ACTION"] = "TRACK_ACTION";
  ExtensionRequestType["ERROR_EVENT"] = "ERROR_EVENT";
  ExtensionRequestType["INVOKE_EXTERNAL_API"] = "INVOKE_EXTERNAL_API";
  ExtensionRequestType["EXTENSION_UNLOADED"] = "EXTENSION_UNLOADED";
  ExtensionRequestType["SPARTAN_LOGOUT"] = "SPARTAN_LOGOUT";
})(ExtensionRequestType || (ExtensionRequestType = {}));

export var ApiVersion;

(function (ApiVersion) {
  ApiVersion["sdk31"] = "3.1";
  ApiVersion["sdk40"] = "4.0";
})(ApiVersion || (ApiVersion = {}));

export var FetchResponseBodyType;

(function (FetchResponseBodyType) {
  FetchResponseBodyType["json"] = "json";
  FetchResponseBodyType["text"] = "text";
})(FetchResponseBodyType || (FetchResponseBodyType = {}));

export var ExtensionNotificationType;

(function (ExtensionNotificationType) {
  ExtensionNotificationType["ROUTE_CHANGED"] = "ROUTE_CHANGED";
  ExtensionNotificationType["INITIALIZE"] = "INITIALIZE";
})(ExtensionNotificationType || (ExtensionNotificationType = {}));
