// @looker/sdk/lib/3.1/models.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WeekStartDay = exports.UserAttributeFilterTypes = exports.SupportedVisualizationFormattings = exports.SupportedFormattings = exports.SupportedFormats = exports.SupportedDownloadSettings = exports.SupportedActionTypes = exports.SslVersion = exports.ResultFormat = exports.PullRequestMode = exports.PermissionType = exports.Name = exports.Format = exports.FillStyle = exports.DependencyStatus = exports.Category = exports.Align = void 0;
var Align;
exports.Align = Align;

(function (Align) {
  Align["left"] = "left";
  Align["right"] = "right";
})(Align || (exports.Align = Align = {}));

var Category;
exports.Category = Category;

(function (Category) {
  Category["parameter"] = "parameter";
  Category["filter"] = "filter";
  Category["measure"] = "measure";
  Category["dimension"] = "dimension";
})(Category || (exports.Category = Category = {}));

var DependencyStatus;
exports.DependencyStatus = DependencyStatus;

(function (DependencyStatus) {
  DependencyStatus["lock_optional"] = "lock_optional";
  DependencyStatus["lock_required"] = "lock_required";
  DependencyStatus["lock_error"] = "lock_error";
  DependencyStatus["install_none"] = "install_none";
})(DependencyStatus || (exports.DependencyStatus = DependencyStatus = {}));

var FillStyle;
exports.FillStyle = FillStyle;

(function (FillStyle) {
  FillStyle["enumeration"] = "enumeration";
  FillStyle["range"] = "range";
})(FillStyle || (exports.FillStyle = FillStyle = {}));

var Format;
exports.Format = Format;

(function (Format) {
  Format["topojson"] = "topojson";
  Format["vector_tile_region"] = "vector_tile_region";
})(Format || (exports.Format = Format = {}));

var Name;
exports.Name = Name;

(function (Name) {
  Name["day"] = "day";
  Name["hour"] = "hour";
  Name["minute"] = "minute";
  Name["second"] = "second";
  Name["millisecond"] = "millisecond";
  Name["microsecond"] = "microsecond";
  Name["week"] = "week";
  Name["month"] = "month";
  Name["quarter"] = "quarter";
  Name["year"] = "year";
})(Name || (exports.Name = Name = {}));

var PermissionType;
exports.PermissionType = PermissionType;

(function (PermissionType) {
  PermissionType["view"] = "view";
  PermissionType["edit"] = "edit";
})(PermissionType || (exports.PermissionType = PermissionType = {}));

var PullRequestMode;
exports.PullRequestMode = PullRequestMode;

(function (PullRequestMode) {
  PullRequestMode["off"] = "off";
  PullRequestMode["links"] = "links";
  PullRequestMode["recommended"] = "recommended";
  PullRequestMode["required"] = "required";
})(PullRequestMode || (exports.PullRequestMode = PullRequestMode = {}));

var ResultFormat;
exports.ResultFormat = ResultFormat;

(function (ResultFormat) {
  ResultFormat["inline_json"] = "inline_json";
  ResultFormat["json"] = "json";
  ResultFormat["json_detail"] = "json_detail";
  ResultFormat["json_fe"] = "json_fe";
  ResultFormat["csv"] = "csv";
  ResultFormat["html"] = "html";
  ResultFormat["md"] = "md";
  ResultFormat["txt"] = "txt";
  ResultFormat["xlsx"] = "xlsx";
  ResultFormat["gsxml"] = "gsxml";
})(ResultFormat || (exports.ResultFormat = ResultFormat = {}));

var SslVersion;
exports.SslVersion = SslVersion;

(function (SslVersion) {
  SslVersion["TLSv1_1"] = "TLSv1_1";
  SslVersion["SSLv23"] = "SSLv23";
  SslVersion["TLSv1_2"] = "TLSv1_2";
})(SslVersion || (exports.SslVersion = SslVersion = {}));

var SupportedActionTypes;
exports.SupportedActionTypes = SupportedActionTypes;

(function (SupportedActionTypes) {
  SupportedActionTypes["cell"] = "cell";
  SupportedActionTypes["query"] = "query";
  SupportedActionTypes["dashboard"] = "dashboard";
  SupportedActionTypes["none"] = "none";
})(SupportedActionTypes || (exports.SupportedActionTypes = SupportedActionTypes = {}));

var SupportedDownloadSettings;
exports.SupportedDownloadSettings = SupportedDownloadSettings;

(function (SupportedDownloadSettings) {
  SupportedDownloadSettings["push"] = "push";
  SupportedDownloadSettings["url"] = "url";
})(SupportedDownloadSettings || (exports.SupportedDownloadSettings = SupportedDownloadSettings = {}));

var SupportedFormats;
exports.SupportedFormats = SupportedFormats;

(function (SupportedFormats) {
  SupportedFormats["txt"] = "txt";
  SupportedFormats["csv"] = "csv";
  SupportedFormats["inline_json"] = "inline_json";
  SupportedFormats["json"] = "json";
  SupportedFormats["json_label"] = "json_label";
  SupportedFormats["json_detail"] = "json_detail";
  SupportedFormats["json_detail_lite_stream"] = "json_detail_lite_stream";
  SupportedFormats["xlsx"] = "xlsx";
  SupportedFormats["html"] = "html";
  SupportedFormats["wysiwyg_pdf"] = "wysiwyg_pdf";
  SupportedFormats["assembled_pdf"] = "assembled_pdf";
  SupportedFormats["wysiwyg_png"] = "wysiwyg_png";
  SupportedFormats["csv_zip"] = "csv_zip";
})(SupportedFormats || (exports.SupportedFormats = SupportedFormats = {}));

var SupportedFormattings;
exports.SupportedFormattings = SupportedFormattings;

(function (SupportedFormattings) {
  SupportedFormattings["formatted"] = "formatted";
  SupportedFormattings["unformatted"] = "unformatted";
})(SupportedFormattings || (exports.SupportedFormattings = SupportedFormattings = {}));

var SupportedVisualizationFormattings;
exports.SupportedVisualizationFormattings = SupportedVisualizationFormattings;

(function (SupportedVisualizationFormattings) {
  SupportedVisualizationFormattings["apply"] = "apply";
  SupportedVisualizationFormattings["noapply"] = "noapply";
})(SupportedVisualizationFormattings || (exports.SupportedVisualizationFormattings = SupportedVisualizationFormattings = {}));

var UserAttributeFilterTypes;
exports.UserAttributeFilterTypes = UserAttributeFilterTypes;

(function (UserAttributeFilterTypes) {
  UserAttributeFilterTypes["advanced_filter_string"] = "advanced_filter_string";
  UserAttributeFilterTypes["advanced_filter_number"] = "advanced_filter_number";
  UserAttributeFilterTypes["advanced_filter_datetime"] = "advanced_filter_datetime";
  UserAttributeFilterTypes["string"] = "string";
  UserAttributeFilterTypes["number"] = "number";
  UserAttributeFilterTypes["datetime"] = "datetime";
  UserAttributeFilterTypes["relative_url"] = "relative_url";
  UserAttributeFilterTypes["yesno"] = "yesno";
  UserAttributeFilterTypes["zipcode"] = "zipcode";
})(UserAttributeFilterTypes || (exports.UserAttributeFilterTypes = UserAttributeFilterTypes = {}));

var WeekStartDay;
exports.WeekStartDay = WeekStartDay;

(function (WeekStartDay) {
  WeekStartDay["monday"] = "monday";
  WeekStartDay["tuesday"] = "tuesday";
  WeekStartDay["wednesday"] = "wednesday";
  WeekStartDay["thursday"] = "thursday";
  WeekStartDay["friday"] = "friday";
  WeekStartDay["saturday"] = "saturday";
  WeekStartDay["sunday"] = "sunday";
})(WeekStartDay || (exports.WeekStartDay = WeekStartDay = {}));
